misk-testing/src/main/kotlin/misk/web/WebActionRegistrationTesting.kt

```diff
@@ -0,0 +1,181 @@
+package misk.web
+
+import com.google.inject.Injector
+import com.google.inject.Key
+import com.google.inject.TypeLiteral
+import io.github.classgraph.ClassGraph
+import misk.web.actions.WebAction
+import misk.web.actions.WebActionEntry
+import org.assertj.core.api.Assertions.assertThat
+import java.lang.reflect.Modifier
+import kotlin.reflect.KClass
+
+/**
+ * Test utilities for verifying that all WebAction implementations are properly registered.
+ *
+ * This utility helps catch a common mistake: implementing a WebAction but forgetting to register
+ * it via `WebActionModule.create<YourAction>()`. Without registration, the action won't be exposed
+ * as an HTTP endpoint.
+ *
+ * ## Usage
+ *
+ * ```kotlin
+ * @Test
+ * fun allWebActionsAreRegistered() {
+ *   val injector = // create your injector
+ *   WebActionRegistrationTesting.assertAllWebActionsRegistered(
+ *     injector,
+ *     WebActionRegistrationTesting.Options(
+ *       basePackages = listOf("com.example.myservice"),
+ *     )
+ *   )
+ * }
+ * ```
+ *
+ * Or use [AbstractWebActionRegistrationTest] for a simpler base class approach.
+ */
+object WebActionRegistrationTesting {
```

Comment:
```suggestion
object WebActionRegistrationTester {
```

I like tester or verifier vs. a passive verb as the object suffix